### ISSUE TITLE
Testing fails for View Components due to component_controller

### ISSUE THREAD

**Issue by cj** [joshleblanc/view_component_reflex#44]
When doing a simple test i.e.

```ruby
# frozen_string_literal: true

require "test_helper"

class FormComponent::BaseTest < ViewComponent::TestCase
  test "rendering form" do
    render_inline(FormComponent::Base.new("/"))
  end
end
```

This error is thrown:

```
14:01:16 guard.1 | Minitest::UnexpectedError:         NoMethodError: undefined method `id' for {}:Hash
14:01:16 guard.1 |             app/components/form_component/base.html.erb:1:in `call'
14:01:16 guard.1 |             (eval):3:in `render_template_for'
14:01:16 guard.1 |             test/components/form_component/base_test.rb:15:in `render_inline'
14:01:16 guard.1 |             test/components/form_component/base_test.rb:24:in `block in <class:BaseTest>'
```

It is failing on `component_controller`.  Currently I have it stubbed for tests by doing the following:

```ruby
AppComponent::Base.class_eval do
  def component_controller(*_args, &block)
    block.call
  end
end
```

I am going to try and track down the error to see if there is a better solution.

**Comment by cj** [comment]
It most likely is `request.session.id`

**Comment by cj** [comment]
Ok so here is the fix:

```ruby
  def render_inline(component, **args, &block)
    controller.stubs(:request).returns(OpenStruct.new({
      env: {},
      session: OpenStruct.new,
    }))

    @rendered_component =
      if Rails.version.to_f >= 6.1
        controller.view_context.render(component, args, &block)
      else
        controller.view_context.render_component(component, &block)
      end

    Nokogiri::HTML.fragment(@rendered_component)
  end
```

**Comment by joshleblanc** [comment]
This should be resolved in 3.0.2

Thanks for the report!
